improver_tests/nowcasting/utilities/test_FillRadarHoles.py

```diff
@@ -0,0 +1,128 @@
+"""Unit tests for the FillRadarHoles plugin"""
+
+import numpy as np
+import pytest
+from iris.cube import Cube
+from numpy.ma import MaskedArray
+
+from improver.nowcasting.utilities import FillRadarHoles
+
+from ...set_up_test_cubes import set_up_variable_cube
+
+
+def create_masked_rainrate_data():
+    """Create a masked data array of rain rates in mm/h"""
+    nonzero_data = np.array(
+        [
+            [0.03, 0.1, 0.1, 0.1, 0.03],
+            [0.1, 0.2, 0.2, np.nan, 0.1],
+            [0.2, 0.5, np.nan, np.nan, 0.2],
+            [0.1, 0.5, np.nan, np.nan, 0.1],
+            [0.03, 0.2, 0.2, 0.1, 0.03],
+        ]
+    )
+    data = np.zeros((16, 16), dtype=np.float32)
+    data[5:12, 5:12] = np.full((7, 7), 0.03, dtype=np.float32)
+    data[6:11, 6:11] = nonzero_data.astype(np.float32)
+    mask = np.where(np.isfinite(data), False, True)
+    return MaskedArray(data, mask=mask)
+
+
+RAIN_DATA = create_masked_rainrate_data()
+
+# first case: speckle, rates in mm/h
+RAIN_CUBE = set_up_variable_cube(
+    RAIN_DATA, name="lwe_precipitation_rate", units="mm h-1", spatial_grid="equalarea"
+)
+
+INTERPOLATED_RAIN = RAIN_DATA.copy()
+INTERPOLATED_RAIN.data[7:10, 8:10] = [
+    [0.2, 0.07138586],
+    [0.11366593, 0.09165306],
+    [0.09488520, 0.07650946],
+]
+INTERPOLATED_RAIN.mask = np.full(INTERPOLATED_RAIN.shape, False)
+
+# second case: speckle, rates in m/s
+MS_RAIN_CUBE = RAIN_CUBE.copy()
+MS_RAIN_CUBE.convert_units("m s-1")
+
+MS_INTERPOLATED_RAIN = INTERPOLATED_RAIN / (3600.0 * 1000.0)
+
+# third case: widespread mask
+MASKED_RAIN_CUBE = RAIN_CUBE.copy()
+MASKED_RAIN_CUBE.data[:11, :11] = np.nan
+MASKED_RAIN_CUBE.data.mask = np.where(
+    np.isfinite(MASKED_RAIN_CUBE.data.data), False, True
+)
+
+# set up alternate test cases for which interpolation should ("speckle_*")
+# and should not ("masked") be triggered
+CASES = ["speckle_mmh", "speckle_ms", "masked"]
+INPUT_CUBES = {
+    "speckle_mmh": RAIN_CUBE,
+    "speckle_ms": MS_RAIN_CUBE,
+    "masked": MASKED_RAIN_CUBE,
+}
+OUTPUT_DATA = {
+    "speckle_mmh": INTERPOLATED_RAIN,
+    "speckle_ms": MS_INTERPOLATED_RAIN,
+    "masked": MASKED_RAIN_CUBE.data.copy(),
+}
+
+PLUGIN = FillRadarHoles()
+
+
+def test_basic():
+    """Test that the plugin returns a masked cube"""
+    result = PLUGIN(RAIN_CUBE)
+    assert isinstance(result, Cube)
+    assert isinstance(result.data, MaskedArray)
+
+
+@pytest.mark.parametrize("case", CASES)
+def test_values(case):
+    """Test that the data and mask contain the expected values"""
+    result = PLUGIN(INPUT_CUBES[case])
+    assert np.allclose(result.data, OUTPUT_DATA[case])
+    assert np.array_equal(result.data.mask, OUTPUT_DATA[case].mask)
+
+
+def test_log_transform_reversability():
+    """Test that data in mm h-1 are reproduced when transforming to
+    and from log rainrates"""
+    log_rr = PLUGIN._rr_to_log_rr(RAIN_CUBE.data)
+    assert all(np.isnan(log_rr[np.where(RAIN_CUBE.data == 0)]))
+    rr = PLUGIN._log_rr_to_rr(log_rr)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable name "rr" doesn't conform to snake_case naming style](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5409292)